### PR TITLE
BaseActiveRecord change to store changed attribute to shrink the memory

### PR DESCRIPTION
**Pull request by tsingsun**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | no

BaseActiveRecord keep the old attribute,i think keep changed attribute will be better,it also trace the changed attribute and shrink the memory useage

### CODE REVIEW

**Comment by samdark** [comment]
Seems to break tests: https://travis-ci.org/yiisoft/yii2/jobs/356687083

**Comment by kushalhalder** [comment]
How much memory are we saving with this?

**Comment by tsingsun** [comment]
@samdark the PR is used in my project,i will check failed tests

**Comment by tsingsun** [comment]
@kushalhalder sorry,i hava not an exact number,but BaseActiveRecord keep current state's data,backup origin data from database source to $_oldAttributes,when fetch from source,two copies data keep in memory.the PR don't persist $_oldAttributes

File: framework/db/BaseActiveRecord.php [hunk 1]
```diff
@@ -92,7 +92,7 @@
      * @var array|null old attribute values indexed by attribute names.
      * This is `null` if the record [[isNewRecord|is new]].
      */
-    private $_oldAttributes;
+    private $_changeAttributes;
```
Comment (Renkas) [inline framework/db/BaseActiveRecord.php]: Why not name it `$_changedAttributes`?

File: framework/db/BaseActiveRecord.php [hunk 2]
```diff
@@ -92,7 +92,7 @@
      * @var array|null old attribute values indexed by attribute names.
```
Comment (Renkas) [inline framework/db/BaseActiveRecord.php]: I think you should also change the documentation part if you change the data this variable contains

**Comment by cebe** [comment]
> sorry,i hava not an exact number,but BaseActiveRecord keep current state's data,backup origin data from database source to $_oldAttributes,when fetch from source,two copies data keep in memory.the PR don't persist $_oldAttributes

dependent on how php handles these internally, they could be pointers to the same data which are changed when data is written (copy on write). We should not make assumptions about this but verify how much memory you safe.

**Comment by tsingsun** [comment]
@cebe give the fate,great!  this pull request will be close

**Pull request closed without merging**